django/db/backends/mysql/operations.py

```diff
@@ -142,6 +143,8 @@ def adapt_datetimefield_value(self, value):
             return None
 
         # MySQL doesn't support tz-aware datetimes
+        if isinstance(value, Combinable):
```

Comment:
I think you might have a point. My logic for putting it there was that that check for time zone is the only reason the short circuit is required, as combinables aren't date times themselves. Happy to move it though.